crates/symbolicator/src/cache.rs

```diff
@@ -285,48 +285,69 @@ impl Cache {
 
     pub fn cleanup(&self) -> Result<()> {
         tracing::info!("Cleaning up cache: {}", self.name);
-        let cache_dir = self.cache_dir.clone().ok_or_else(|| {
+        let cache_dir = self.cache_dir.as_ref().ok_or_else(|| {
             anyhow!("no caching configured! Did you provide a path to your config file?")
         })?;
 
-        let mut directories = vec![cache_dir];
-        while !directories.is_empty() {
-            let directory = directories.pop().unwrap();
+        self.cleanup_directory_recursive(cache_dir)?;
 
-            let entries = match catch_not_found(|| read_dir(directory))? {
-                Some(x) => x,
-                None => {
-                    tracing::warn!("Directory not found");
-                    return Ok(());
-                }
-            };
+        Ok(())
+    }
+
+    /// Cleans up the directory recursively, returning `true` if the directory is left empty after cleanup.
+    fn cleanup_directory_recursive(&self, directory: &Path) -> Result<bool> {
+        let entries = match catch_not_found(|| read_dir(directory))? {
```

Comment:
`s/catch/cache`?

Reply:
This is correct, it catches not found errors, turning them into `None`.